src/Service/Parser/TupleColumnArrayValueParser.php

```diff
@@ -0,0 +1,43 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\ApiFilter\Service\Parser;
+
+use Lmc\ApiFilter\Assertion;
+use Lmc\ApiFilter\Constant\Filter;
+use MF\Collection\Immutable\Tuple;
+
+class TupleColumnArrayValueParser extends AbstractParser
+{
+    public function supports(string $rawColumn, $rawValue): bool
+    {
+        return $this->isTuple($rawColumn) && is_array($rawValue);
+    }
+
+    public function parse(string $rawColumn, $rawValue): iterable
+    {
+        $columns = Tuple::parse($rawColumn)->toArray();
+        $columnsCount = count($columns);
+
+        foreach ($rawValue as $filter => $tupleValue) {
+            Assertion::notSame(Filter::IN, $filter, 'Tuples are not allowed in IN filter.');
+            $values = $this->parseValue($tupleValue, $columnsCount);
+
+            foreach ($columns as $column) {
+                Assertion::false(
```

Comment:
Considering method `isColumnWithFilter` is used only here, it may be more readable and cohesive to move the logic to own domain assertion and call it cleanly like this: `Assertion::isColumnWithFilter($column);`